### ISSUE TITLE
when config is loaded from different sources, List properties aren't correctly loaded in some cases

### ISSUE THREAD

**Issue by smartrics** [tonybaines/gestalt#20]
This class reproduces the problem

```
public class GestaltSpike {

    public interface AppConfigSample {

        List<PropertySample> getProperties();
    }

    public interface PropertySample {
        String getKey();

        String getValue();
    }

    static String xml = "<AppConfigSample>\n" +
            "  <properties>\n" +
            "    <propertySample>\n" +
            "      <value>/a/b</value>\n" +
            "      <key>webservices.url</key>\n" +
            "    </propertySample>\n" +
            "    <propertySample>\n" +
            "      <value>pong</value>\n" +
            "      <key>ping</key>\n" +
            "    </propertySample>\n" +
            "  </properties>\n" +
            "</AppConfigSample>\n";

    public static void main(String[] args) {
        AppConfigSample config = Configurations.definedBy(AppConfigSample.class)
                .fromProperties(new Properties())
                .fromXml(new ByteArrayInputStream(xml.getBytes()))
                .without(Configurations.Feature.ExceptionOnNullValue).done();
        System.out.println(Configurations.toXml(config, AppConfigSample.class));

   }
}
```

If run as is above this is the output which is incorrect

```
<AppConfigSample>
  <properties />
</AppConfigSample>
```

If the line .fromProperties(new Properties()) is removed, then hte output is correct:

```
<AppConfigSample>
  <properties>
    <propertySample>
      <value>/a/b</value>
      <key>webservices.url</key>
    </propertySample>
    <propertySample>
      <value>pong</value>
      <key>ping</key>
    </propertySample>
  </properties>
</AppConfigSample>
```

**Comment by tonybaines** [comment]
The bug is that accessing a list type e.g. `properties` from a `java.util.Properties` instance should return `null` if there's no associated values (currently returning an empty `List`)

**Comment by tonybaines** [comment]
Fixed in 1.0.3
